library/ecp_invasive.h

```diff
@@ -31,8 +31,30 @@
 #include "bignum_mod.h"
 #include "mbedtls/ecp.h"
 
+#ifndef MBEDTLS_ECP_WITH_MPI_UINT
```

Comment:
I am not sure I follow the logic.

    If (MBEDTLS_ECP_WITH_MPI_STRUCT)
         define MBEDTLS_ECP_WITH_MPI_UINT
         undefine MBEDTLS_ECP_WITH_MPI_UINT
         define MBEDTLS_ECP_WITH_MPI_STRUCT

Is that file aimed at testing the sanity of the compile time parameters? 

With the current unit compilation order, 

    ecp.c
    ecp_new.c
    ecp_curves.c
    ecp_curves_new.c
    entropy.c

the `MBEDTLS_ECP_WITH_MPI_STRUCT` will always be defined and the `MBEDTLS_ECP_WITH_MPI_UINT` will only be if passed it as a compile time flag.

I think this blocked is aimed at working around that, but could be masking that behaviour when performing release/production builds(since we enable the TEST_HOOKS to test ecp).That workaround works in the CI but ma not always work.

Reply:
Gabor's changes, and this was improved.